### ISSUE TITLE
Safari layout issues

### ISSUE THREAD

**Issue by minrk** [jupyter/jupyterlab#749]
The existing notebook officially supports reasonably current Safari, Chrome, and Firefox, but JupyterLab has various layout issues on Safari. We need to work these out before we can get to a stable release of JupyterLab, so that it is at least usable on the officially supported browsers.

Menu items don't layout correctly in stable Safari:

<img width="343" alt="screen shot 2016-08-24 at 23 56 34" src="https://cloud.githubusercontent.com/assets/151929/17949616/7e8a0012-6a56-11e6-8b53-c16135a69f86.PNG">

but this is fixed in the Tech Preview, which I suppose is why #109 was closed as wontfix. I think that's unfortunate but could be okay given the timeline for JupyterLab getting a stable release. It's not the only issue, though. Some context-menu items appear to have issues, even in the Tech Preview:

<img width="360" alt="screen shot 2016-08-24 at 23 56 27" src="https://cloud.githubusercontent.com/assets/151929/17949708/05b641f4-6a57-11e6-8dad-0ae1c087817f.PNG">

Opening this context menu gives the following errors, which I'm guessing are related:

<img width="1217" alt="screen shot 2016-08-25 at 00 05 17" src="https://cloud.githubusercontent.com/assets/151929/17950093/49518656-6a59-11e6-9574-3c8cb85185b5.PNG">

I'll see if I can find some time to track down the right fixes, but I'm guessing there are people with more relevant expertise than me.

**Comment by jasongrout** [comment]
Thanks for looking into this!

**Comment by ellisonbg** [comment]
Moving this up 0.9

**Comment by blink1073** [comment]
IIUC this would require fundamental changes to phosphor that are the result of Safari not following the browser spec.

**Comment by blink1073** [comment]
cc @sccolbert

**Comment by minrk** [comment]
Given that the tech preview (due for release this Fall) does fix the CSS issue described in #109, I don't think it's critical to fix that one, but I do think it's worth spending some time to see what it might require. phosphorjs/phosphor#137 should fix the submenu failure.

**Comment by blink1073** [comment]
Great, there's hope yet!

**Comment by sccolbert** [comment]
The layout issues could possibly be solved with vendor-prefixing the flex layout. But I'm not sure. Can someone tag safari-specific issues with a "safari" tag so that I can look into them?

**Comment by blink1073** [comment]
@sccolbert, done.

**Comment by blink1073** [comment]
As mentioned in #757, the latest stable release of Safari (10) fixes this issue.

**Comment by blink1073** [comment]
Also confirmed that context menus work in the current master with the stable chrome.
